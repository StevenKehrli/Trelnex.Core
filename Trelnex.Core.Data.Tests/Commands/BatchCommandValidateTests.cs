using FluentValidation;
using Snapshooter.NUnit;

namespace Trelnex.Core.Data.Tests.Commands;

public class BatchCommandValidateTests
{
    private readonly string _typeName = "test-item";

    [Test]
    public async Task BatchCommandValidate_ValidateAsync()
    {
        var validator = new InlineValidator<TestItem>();
        validator.RuleFor(k => k.PublicMessage).NotEmpty();
        validator.RuleFor(k => k.PrivateMessage).NotEmpty();

        var id = Guid.NewGuid().ToString();
        var partitionKey = Guid.NewGuid().ToString();

        var requestContext = TestRequestContext.Create();

        // create our command provider
        var factory = await InMemoryCommandProviderFactory.Create();

        var commandProvider = factory.Create<ITestItem, TestItem>(
                typeName: _typeName,
                validator: validator);

        var batchCommand = commandProvider.Batch();

        // batch a create command
        var createCommand1 = commandProvider.Create(
            id: id,
            partitionKey: partitionKey);

        createCommand1.Item.PrivateMessage = "Private #1";

        batchCommand.Add(createCommand1);

        // batch a create command
        var createCommand2 = commandProvider.Create(
            id: id,
            partitionKey: partitionKey);

        createCommand2.Item.PublicMessage = "Public #2";

        batchCommand.Add(createCommand2);

        // batch a create command
        var createCommand3 = commandProvider.Create(
            id: id,
            partitionKey: partitionKey);

        createCommand3.Item.PrivateMessage = "Private #3";
        createCommand3.Item.PublicMessage = "Public #3";

        batchCommand.Add(createCommand3);

        // validate
        var validationResults = await batchCommand.ValidateAsync(default);

        Snapshot.Match(validationResults);
    }
}
