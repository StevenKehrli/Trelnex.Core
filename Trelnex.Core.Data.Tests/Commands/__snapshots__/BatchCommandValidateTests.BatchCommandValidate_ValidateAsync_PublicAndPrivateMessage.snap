[
  {
    "IsValid": false,
    "Errors": [
      {
        "PropertyName": "PublicMessage",
        "ErrorMessage": "'Public Message' must not be empty.",
        "AttemptedValue": null,
        "CustomState": null,
        "Severity": "Error",
        "ErrorCode": "NotEmptyValidator",
        "FormattedMessagePlaceholderValues": {
          "PropertyName": "Public Message",
          "PropertyValue": null,
          "PropertyPath": "PublicMessage"
        }
      },
      {
        "PropertyName": "PrivateMessage",
        "ErrorMessage": "'Private Message' must not be empty.",
        "AttemptedValue": null,
        "CustomState": null,
        "Severity": "Error",
        "ErrorCode": "NotEmptyValidator",
        "FormattedMessagePlaceholderValues": {
          "PropertyName": "Private Message",
          "PropertyValue": null,
          "PropertyPath": "PrivateMessage"
        }
      }
    ],
    "RuleSetsExecuted": [
      "default"
    ]
  }
]
